src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -297,28 +300,39 @@ private constructor(
 
     @RequiresEdt
     fun createFromState(project: Project, state: ChatState): AgentChatSession {
-      val sessionId = createNewPanel(project) { it.server.chatNew() }
-      val modelFromState = state.model?.let { ChatModel.fromDisplayName(it) }
-      val chatSession = AgentChatSession(project, sessionId, state.internalId!!, modelFromState)
-      state.messages.forEachIndexed { index, message ->
-        val parsed =
-            when (val speaker = message.speaker) {
-              MessageState.SpeakerState.HUMAN -> Speaker.HUMAN
-              MessageState.SpeakerState.ASSISTANT -> Speaker.ASSISTANT
-              else -> error("unrecognized speaker $speaker")
-            }
-        val chatMessage = ChatMessage(speaker = parsed, message.text)
-        chatSession.messages.add(chatMessage)
-        chatSession.chatPanel.addOrUpdateMessage(
-            chatMessage, index, shouldAddBlinkingCursor = false)
-      }
+      val agentChatSession = CompletableFuture<AgentChatSession>()
+      createNewPanel(project) { codyAgent ->
+        val innerSessionId = codyAgent.server.chatNew()
```

Comment:
Maybe just inline that calls?
Now you have problem with tmp names which are kind of duplicated or artificial like `response`:

```
        val mySessionId = codyAgent.server.chatNew().completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
         val chatModels = codyAgent.server.chatModels(ChatModelsParams(mySessionId)).completeOnTimeout(null, 4, TimeUnit.SECONDS).get()
      ```